src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1252,9 +1252,12 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                                                        out okToDowngradeToNeither);
                 }
 
+                var type1Normalized = type1.NormalizeTaskTypes(Compilation);
+                var type2Normalized = type2.NormalizeTaskTypes(Compilation);
```

Comment:
Good catch. If `Task` and `Task<T>` are not available, we are currently substituting an error type. I've changed `NormalizeTaskTypes` to skip substitution in that case but that means we will not consider two different task-like types as equivalent if the well known types are missing.